packages/blade/src/components/Card/_decisions/decisions.md

```diff
@@ -0,0 +1,512 @@
+- Why keep the `titlePrefix` flexible? When from the design side we have constraints that the size & color of the Icon cannot be changed? 
+- With `titlePrefix` users will need to pass the `color` & `size` prop to the Icon since color prop is mandatory & size is needed to be `xlarge`
+
+We discussed on why we approached this API like:  
+
+<details>
+<summary><b>Q:</b> This API is too constraints in some cases & too flexible in other cases</summary>
+
+----
+**Ans:** 
+There are few sections in the Card which doesn’t need to be changed like header/footer title, subtitle, footer’s actions, and other sections which we decided to keep a bit more flexible to accommodate future usecases like: What if in future instead of counter we need to add a badge? Or there could be cases were we need to add an Avatar in the titlePrefix thus we decided to keep those parts more flexible. 
+
+</details>
+
+<details>
+<summary><b>Q:</b> Why keep the titlePrefix flexible? When from the design side we have constraints that the Size & color of the Icon cannot be changed?</summary>
+
+----
+**Ans:** 
+For now we only have usecase of an icon there but neither development side nor design side knows for sure what other usecases there might arise, so we decided to keep it flexible. 
+
+</details>
+
+<details>
+<summary><b>Q:</b> With titlePrefix users will need to pass the color & size prop to the Icon since color prop is mandatory & size is needed to be “xlarge”</summary>
+
+----
+**Ans:** 
+This was a valid concern with the API which we all agreed that needs to be fixed, since not only does it have a footgun, the consumer also needs to know the exact size & color token to be used in there which is not ideal. 
+
+</details>
+
+
+**API 5:** Semi-Flexible Compound API:  
+
+
+**Pros:** 
+- Addresses the issue with users needing to pass size & color to icon
+
+**Cons:** 
+- There will be a lot of complex runtime checks we need to do.
+We also need to ensure the ordering of the JSX elements if users change the order, which will lead to more runtime hacks. 
+
+```jsx
+<Card.Header>
+  <Card.HeaderLeading>
+    <CardHeader.Icon icon={Clock} />
+    <Card.HeaderTitle title="" subtitle=""  />
+    <CardHeader.Counter />
+  </Card.HeaderLeading>
+  <Card.HeaderTrailing>
+    <Badge />
+  </Card.HeaderTrailing>
+</Card.Header>
+```
+
+**API 6:** Constrained Prop Based: 
+
+**Pros:** 
+- Gives us total control of how we want to structure the Card
+No footguns as such
+
+**Cons:** 
+- Prop based API seems a bit too restrictive
+- Hard to extend later on for future usecases, imagine if we want to add Avatar we need to add `prefixAvatar` prop for it
+- Passing props to counter needs another prop `counterIntent`
+
+```jsx
+<Card>
+  <Card.Header
+    title=""
+    subtitle=""
+    prefixIcon={CloseIcon}
+    prefixAvatar={}
+    counter={12}
+    counterIntent=""
+    trailingVisual={<Badge />}
+  />
+</Card>
+```
+
+**API 7: Final API**
```

Comment:
now API 8? where we move away from the static attributes name checking? 😄

Reply:
😆 haha yeah!